core/src/main/scala/chisel3/ChiselEnum.scala

```diff
@@ -159,6 +159,8 @@ abstract class EnumType(private[chisel3] val factory: ChiselEnum, selfAnnotating
         case (name, v: Vec[_]) if this.typeEquivalent(v.sample_element) => Seq(Seq(name))
         case (name, b2: Bundle) => enumFields(b2).map(name +: _)
       }.flatten.toSeq
+    case _ =>
+      throw new Exception(s"Match Error: d=$d")
   }
```

Comment:
This one looks like it could be a really bug, instead of Bundle above it probably should be `Record`.

Reply:
Changed to bundle and non-exhaustive warning went away